src/Temporalio/Client/Schedules/ScheduleActionStartWorkflow.cs

```diff
@@ -0,0 +1,182 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Google.Protobuf.WellKnownTypes;
+using Temporalio.Api.Common.V1;
+using Temporalio.Converters;
+
+namespace Temporalio.Client.Schedules
+{
+    /// <summary>
+    /// Schedule action to start a workflow. Instead of the constructor, most users should use the
+    /// <c>Create</c> static method to create a typed workflow invocation.
+    /// </summary>
+    /// <param name="Workflow">Workflow type name.</param>
+    /// <param name="Args">Arguments for the workflow. Note, when fetching this from the server,
+    /// the every value here is an instance of <see cref="IEncodedRawValue" />.</param>
+    /// <param name="Options">Start workflow options. ID and TaskQueue are required.</param>
+    /// <param name="Headers">Headers sent with each workflow scheduled.</param>
+    public record ScheduleActionStartWorkflow(
+        string Workflow,
+        IReadOnlyCollection<object?> Args,
+        WorkflowOptions Options,
+        IReadOnlyDictionary<string, Payload>? Headers = null) : ScheduleAction
+    {
+        /// <summary>
+        /// Create a scheduled action that starts a workflow.
+        /// </summary>
+        /// <typeparam name="TResult">Result type of the workflow.</typeparam>
+        /// <param name="workflow">Workflow run method.</param>
+        /// <param name="options">Start workflow options. ID and TaskQueue are required.</param>
+        /// <returns>Start workflow action.</returns>
+        public static ScheduleActionStartWorkflow Create<TResult>(
+            Func<Task<TResult>> workflow, WorkflowOptions options) =>
```

Comment:
might be worth calling out some fields on `WorkflowOptions ` cannot be set for a schedule and will not be used.

Reply:
True, will do.

Reply:
Done (and added error checks for them)